app/src/main/java/com/example/poste/activities/DashboardActivity.java

```diff
@@ -195,30 +239,54 @@ public void onClick(DialogInterface dialog, int which) {
                 String itemName = editTextItemName.getText().toString().trim();
                 String itemLink = editTextItemLink.getText().toString().trim();
                 String selectedFolderName = spinnerNewPostFolder.getSelectedItem().toString();
-                Integer selectedFolderId = folderIdNameMap.get(selectedFolderName);
-
-                if (itemName == null || itemName == "" || itemLink == null || itemLink == "") {
-                    Toast.makeText(DashboardActivity.this, "Cannot create post, please enter name and link", Toast.LENGTH_LONG).show();
+                String selectedFolderId = folderNameToIdMap.get(selectedFolderName);
+
+                if (itemName == "" || itemLink == null || itemLink == "") {
+                    Toast.makeText(
+                            DashboardActivity.this,
+                            "Cannot create post, please enter name and link",
+                            Toast.LENGTH_LONG
+                    ).show();
                     return;
                 }
 
                 if (!itemLink.matches("^((http|https):\\/\\/)(www.)?[a-zA-Z0-9@:%._\\\\+~#?&/=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%._\\\\+~#?&/=]*)$")) {
```

Comment:
(Out of scope) Might be really helpful to move this to utilility functions.

Reply:
Although, if I'm not wrong this only accepts URLs that only begin with www. So some links might incorrectly deemed invalid. For example, when sharing from different apps some apps do it weird, such as facebook might do a m.facebook.com/..... or certain ones can do app.xyz.com/....

Reply:
For now, removing the validator altogether seems acceptable to me. 
but opening up the domain to anything and everything might be _slightly_ out of scope given that the layouts and everything else explicitly ask the user to put down a URL.

Reply:
No longer used since we moved to NewPostActivity altogether